packages/compiler/src/core/checker.ts

```diff
@@ -2737,6 +2738,34 @@ export function createChecker(program: Program): Checker {
         }
       }
 
+      function checkForSelfReference(node: ModelSpreadPropertyNode, parent: ModelStatementNode) {
+        let targetName = "";
+        let parentName = "";
+
+        let targetRef = resolveTypeReferenceSym(node.target, undefined);
+        if (targetRef && targetRef.flags & SymbolFlags.Alias) {
```

Comment:
Do you actually need to do that, doesn't the method above already resolve the alias for you?

Reply:
It does not. It will reference Bar until I resolve the aliased symbol, then it says Foo.

Reply:
ah, yeah I see, got confused, only resolve the base symbol full when doing a member reference

Reply:
then as a follow up wouldnt' just comparing for equality of the resolved symbol vs the model symbol work?

Reply:
After moving this into checkSpreadProperty, the comparison is much easier so this isn't needed.